### PR TITLE
Make missing credential data no longer block slave_setup

### PR DESCRIPTION
**Pull request by timopollmeier**

### CODE REVIEW

File: src/manage.c [hunk 1]
```diff
@@ -2978,45 +2982,48 @@ slave_setup (gvm_connection_t *connection, const char *name, task_t task,
                   || privacy_algorithm == NULL)
                 {
                   cleanup_iterator (&credentials);
-                  goto fail_esxi_credential;
+                  global_slave_esxi_credential_uuid = NULL;
```
Comment (mattmundell) [inline src/manage.c]: Should be global_slave_snmp_credential_uuid, I think.

File: src/manage.c [hunk 2]
```diff
@@ -2977,26 +2977,46 @@ slave_setup (gvm_connection_t *connection, const char *name, task_t task,
...
+                }
+              if (((privacy_password && strcmp (privacy_password, ""))
+                    || (privacy_algorithm && strcmp (privacy_algorithm, "")))
+                  && (password == NULL
+                      || auth_algorithm == NULL
+                      || privacy_password == NULL
+                      || privacy_algorithm == NULL))
+                {
+                  cleanup_iterator (&credentials);
+                  slave_esxi_credential_uuid = NULL;
```
Comment (mattmundell) [inline src/manage.c]: Also should be snmp.

File: src/manage.c [hunk 3]
```diff
@@ -2977,26 +2977,46 @@ slave_setup (gvm_connection_t *connection, const char *name, task_t task,
...
                   global_slave_esxi_credential_uuid = NULL;
                   g_warning ("Could not create slave SNMP credential"
-                             " (missing community, login, password,"
-                             " auth_algorithm, privacy_password"
-                             " or privacy_algorithm)."
+                             " (auth_algorithm must be set if password is"
+                             " given)."
+                             " Continuing without credential.");
+                }
+              if (((privacy_password && strcmp (privacy_password, ""))
```
Comment (mattmundell) [inline src/manage.c]: Surely this should be an "else if", because the if above says it will fail if true.

File: src/manage.c (hunk 2 above)
Comment (timopollmeier) [reply src/manage.c]: I've changed this and the above in d01d4f3.

File: src/manage.c (hunk 3 above)
Comment (timopollmeier) [reply src/manage.c]: You're right. I've changed it in d01d4f3 as well.

**Pull request merged**